src/onegov/event/utils/guidle.py

```diff
@@ -86,15 +86,12 @@ def get(
             return self.cleaner.clean(result) if result else ''
 
     def join(self, texts: 'Sequence[str]', joiner: str = ', ') -> str:
-        """ Joins a set of text, skips duplicate and empty texts. """
-
-        # FIXME: It seems weird that duplicates get skipped initially
-        #        and only the last one shows up, if we don't care about
-        #        order, why don't we just use a set to deduplicate?
-        return joiner.join(
-            text for index, text in enumerate(texts)
-            if text and text not in texts[index + 1:]
-        )
+        """ Joins a set of text, skips duplicate and empty texts while
+        preserving the order. """
+
+        texts = [text for text in texts if text]
+        retval = joiner.join(sorted(set(texts), key=texts.index))
```

Comment:
```suggestion
        deduplicated = {text for text in texts if text}
        retval = joiner.join(sorted(deduplicated, key=texts.index))
```

Directly use a set comprehension. Also do we really need to preserve sort order? `texts.index` is pretty slow as far as key lookups go for sorting. If we wanted to preserve the original order we should do something like this:
```suggestion
        seen = set()
        retval = joiner.join(
            text for text in texts
            if text and text not in seen and seen.add(text) is None
        )
```

Or maybe actually use a loop to avoid the hacky `seen.add` inside the comprehension
```suggestion
        deduplicated = []
        for text in texts:
            if text and text not in deduplicated:
                deduplicated.append(text)

        retval = joiner.join(deduplicated)
```